packages/components/addon/components/hds/link-to/cta.js

```diff
@@ -0,0 +1,179 @@
+//  ██     ██  █████  ██████  ███    ██ ██ ███    ██  ██████
+//  ██     ██ ██   ██ ██   ██ ████   ██ ██ ████   ██ ██
+//  ██  █  ██ ███████ ██████  ██ ██  ██ ██ ██ ██  ██ ██   ███
+//  ██ ███ ██ ██   ██ ██   ██ ██  ██ ██ ██ ██  ██ ██ ██    ██
+//   ███ ███  ██   ██ ██   ██ ██   ████ ██ ██   ████  ██████
+//
+// Notice: in this component we're using directly the styles from the `Hds::Button` component
+// using the `hds-button` class names (and adding a specialized class for the "cta", see below)
+// If you need to change the styling of the `Button` component, remember that this will impact also
+// this component too.
+// If instead you need to change only the styling of the `CTA` component, you can do it
+// in the CSS file using the specialized class declared there.
+// This is NOT a standard approach that we use in the HDS design system implementation, but it's been
+// the least worst option we could find to solve the problem of sharing the exact same style of the
+// `Button (primary)` with other components.
+
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+export const DEFAULT_SIZE = 'medium';
+export const DEFAULT_ICONPOSITION = 'leading';
+export const SIZES = ['small', 'medium', 'large'];
+export const ICONPOSITIONS = ['leading', 'trailing'];
+
+export default class HdsLinkToCtaComponent extends Component {
+  /**
+   * @param text
+   * @type {string}
+   * @description The text of the component. If no text value is defined an error will be thrown.
+   */
+  get text() {
+    let { text } = this.args;
+
+    assert(
+      '@text for "Hds::LinkTo::Cta" must have a valid value',
+      text !== undefined
+    );
+
+    return text;
+  }
+
+  /**
+   * @param size
+   * @type {string}
+   * @default medium
+   * @description The size of the component; acceptable values are `small`, `medium`, and `large`
+   */
+  get size() {
+    let { size = DEFAULT_SIZE } = this.args;
+
+    assert(
+      `@size for "Hds::LinkTo::Cta" must be one of the following: ${SIZES.join(
+        ', '
+      )}; received: ${size}`,
+      SIZES.includes(size)
+    );
+
+    return size;
+  }
+
+  /**
+   * @param icon
+   * @type {string}
+   * @default null
+   * @description The name of the icon to be used.
+   */
+  get icon() {
+    return this.args.icon ?? null;
+  }
+
+  /**
+   * @param iconPosition
+   * @type {string}
```

Comment:
non-blocking question, should be enum (and others)? https://jsdoc.app/tags-enum.html I'm not super familiar with JSDoc, just noticing the diff w/the scrappy page

Reply:
Yeah maybe they could go in the future, one source of truth with the scrappy website "Component API" section, as is already drifting). But I'm not sure how they're used by others TBH. No strong opinion